{"version":3,"file":"exception.test.js","sourceRoot":"","sources":["exception.test.ts"],"names":[],"mappings":";;AAAA,+BAAiC;AACjC,iBAAe;AACf,sCAOsB;AAEtB,KAAK,CAAC,YAAY,EAAE;IAEhB,KAAK,CAAC,sBAAsB,EAAE;QAE1B,IAAI,CAAC,qCAAqC,EAAE;YAExC,IAAI,GAAG,GAAG,IAAI,4BAAoB,CAAC,gBAAgB,CAAC,CAAC;YACrD,IAAI,aAAa,GAAG,GAAG,CAAC,aAAa,CAAC;YACtC,MAAM,CAAC,WAAW,CAAC,GAAG,CAAC,aAAa,EAAE,sBAAsB,CAAC,CAAC;QAClE,CAAC,CAAC,CAAC;IACP,CAAC,CAAC,CAAC;AACP,CAAC,CAAC,CAAC","sourcesContent":["import * as assert from \"assert\";\nimport \"n-ext\";\nimport  \n{\n    ApplicationException,\n    ArgumentException,\n    ArgumentNullException,\n    InvalidArgumentException,\n    InvalidOperationException\n} from \"../src/index\";\n\nsuite(\"exceptions\", () =>\n{\n    suite(\"ApplicationException\", () =>\n    {\n        test(\"should have the right exceptionType\", () =>\n        {\n            let exp = new ApplicationException(\"test exception\");\n            let exceptionType = exp.exceptionType;\n            assert.strictEqual(exp.exceptionType, \"ApplicationException\");\n        });\n    });\n});"]}